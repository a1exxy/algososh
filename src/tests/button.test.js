import React from "react";
import {render, screen, fireEvent, getByTestId} from '@testing-library/react';
import '@testing-library/jest-dom'
import {Button} from "../components/ui/button/button";

describe('1: Тестирование компонента Button (проверяем при помощи снэпшотов корректную отрисовку)', () => {
  it('1.1: кнопки с текстом', () => {
    const testData = '123'
    const {asFragment, getByText} = render(<Button text={testData}/>)
    expect(getByText(testData)).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
  it('1.2: кнопки без текста', () => {
    const testData = ''
    const {asFragment} = render(<Button text={testData}/>)
    expect(asFragment().querySelector('p')).toHaveTextContent(testData)
    expect(asFragment()).toMatchSnapshot()
  })
  it('1.3: заблокированной кнопки', () => {
    const {asFragment} = render(<Button disabled={true}/>)
    expect(screen.getByRole('button')).toHaveAttribute('disabled')
    expect(asFragment()).toMatchSnapshot()
  })
  it('1.4: кнопки с индикацией загрузки', () => {
    const {asFragment} = render(<Button isLoader={true}/>)
    expect(screen.getByRole('img')).toBeInTheDocument()
    expect(screen.getByRole('img')).toHaveClass('loader_icon')
    expect(asFragment()).toMatchSnapshot()
  })

  it('1.5: Проверяем корректность вызова колбека при клике на кнопку', () => {
    const testData = 'GOOD'
    const onClick = () => {
      document.getElementById('result').innerHTML = testData
    }
    const {container} =
      render(<><Button data-testid="btn" onClick={onClick}/><p id="result" data-testid="result"></p></>)
    const btn = getByTestId(container, "btn");
    const result = getByTestId(container, "result");
    fireEvent.click(btn)
    expect(result.textContent).toBe(testData)
    expect(container).toMatchSnapshot()
  })
})