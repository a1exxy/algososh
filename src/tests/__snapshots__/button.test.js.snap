// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1: Тестирование компонента Button (проверяем при помощи снэпшотов корректную отрисовку) 1.1: кнопки с текстом 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      123
    </p>
  </button>
</DocumentFragment>
`;

exports[`1: Тестирование компонента Button (проверяем при помощи снэпшотов корректную отрисовку) 1.2: кнопки без текста 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`1: Тестирование компонента Button (проверяем при помощи снэпшотов корректную отрисовку) 1.3: заблокированной кнопки 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`1: Тестирование компонента Button (проверяем при помощи снэпшотов корректную отрисовку) 1.4: кнопки с индикацией загрузки 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</DocumentFragment>
`;

exports[`1: Тестирование компонента Button (проверяем при помощи снэпшотов корректную отрисовку) 1.5: Проверяем корректность вызова колбека при клике на кнопку 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined false "
    data-testid="btn"
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
  <p
    data-testid="result"
    id="result"
  >
    GOOD
  </p>
</div>
`;
